ops/services/alerts/app_service_metrics/main.tf

```diff
@@ -108,20 +153,26 @@ resource "azurerm_monitor_metric_alert" "http_5xx_errors" {
 
 resource "azurerm_monitor_metric_alert" "http_4xx_errors" {
   name                = "${var.env}-api-4xx-errors"
-  description         = "${local.env_title} HTTP Server 4xx Errors >= 10"
+  description         = "${local.env_title} HTTP Server 4xx Errors (excluding 401s) >= 10"
   resource_group_name = var.rg_name
-  scopes              = [var.app_service_id]
+  scopes              = [var.app_insights_id]
   frequency           = "PT1M"
   window_size         = "PT5M"
   severity            = var.severity
   enabled             = contains(var.disabled_alerts, "http_4xx_errors") ? false : true
 
   criteria {
-    aggregation      = "Average"
-    metric_name      = "Http4xx"
-    metric_namespace = "Microsoft.Web/sites"
+    aggregation      = "Count"
+    metric_name      = "requests/count"
+    metric_namespace = "Microsoft.Insights/Components"
     operator         = "GreaterThanOrEqual"
     threshold        = 10
+
+    dimension {
+      name     = "request/resultCode"
+      operator = "Include"
+      values   = ["400", "403", "404", "410", "429"]
```

Comment:
I wonder if there's a way we can add a warning on backend code that if you add a new error type, you need to include it here. I'd worry about us missing errors otherwise.

Reply:
Yeah not sure how to handle this one. Being able to write a query for this instead would be ideal but the fastest it would run is every 5 minutes. Maybe we could just pre-emptively add codes we might use later (`409`s come to mind)?

Reply:
Sure, that sounds good. Maybe we can add a checklist item too, to make sure alerts won't be disrupted? (though I don't want to overload the checklist either)

Reply:
I think I'm just going to move this to a query-based alert and accept that we only get polling every 5 minutes, since we should be able to switch this to 1 minute soon once the feature is out of preview.